### PR TITLE
mgr/dashboard: RGW user accounts CRUD api's 

### PR DESCRIPTION
**Pull request by nmunet**
Fixes: https://tracker.ceph.com/issues/69110

PR Changes:
Added CRUD & managing quota endpoints for RGW user accounts management

image

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by nmunet** [comment]
jenkins test dashboard cephadm

**Comment by nmunet** [comment]
jenkins test dashboard

**Review by Pegonzal — COMMENTED** [review]
Thanks for the PR @nmunet, overall looks good to me.

Just a couple comments from my side:
- Could you add some tests to cover this endpoints.
- NIT/Suggestion, if the RgwAccounts methods do not need of the instance of the class, could we make those classmethods?

**Comment by Pegonzal** [comment]
@nmunet I was testing this and seems like the cmds are not properly sent?

I'm getting the following error while try the requests with the dashboard swagger

```
400	
Error: Bad Request

Response body

{
  "detail": "Command '['radosgw-admin', '-c', '/etc/ceph/ceph.conf', '-k', '/var/lib/ceph/mgr/ceph-ceph-node-00.nsmsvr/keyring', '-n', 'mgr.ceph-node-00.nsmsvr', 'account', 'create', '--account-name', 'test-account', '--account_id', 'test-id-account', '--email', 'email@test.com']' timed out after 10 seconds",
  "component": "rgw"
}
``

**Comment by nmunet** [comment]
Hey @Pegonzal,
the swagger API are working for me, as I checked you are entering wrong `account-id` as it should be `must be 20 bytes long starting with RGW` and also I am getting proper response for correct payload, FYI

image

Also worked on your suggestions and committed latest.

File: src/pybind/mgr/dashboard/controllers/rgw_iam.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+@APIRouter('rgw/accounts', Scope.RGW)
+@APIDoc("RGW User Accounts API", "RgwUserAccounts")
+class RgwUserAccountsController(RESTController):
+
+    @allow_empty_body
+    def create(self, account_name: Optional[str] = None,
+               account_id: Optional[str] = None, email: Optional[str] = None):
+        accounts = RgwAccounts()
+        return accounts.create_account(account_name, account_id, email)
```
Comment (Pegonzal) [inline src/pybind/mgr/dashboard/controllers/rgw_iam.py:17]:
```suggestion
    def create(self, account_name: Optional[str] = None,
               account_id: Optional[str] = None, email: Optional[str] = None):
        return RgwAccounts.create_account(account_name, account_id, email)
```

**Review by Pegonzal — COMMENTED** [review]
Just tested it again, and it works without issues! Not sure what it went wrong last time.

Anyways, thanks for the follow up @nmunet, one last comment from my side:
Since you went with the @classmethods approach, you no longer need to create an `RgwAccounts` object to call the methods, as now they are directly accessible from the class

File: src/pybind/mgr/dashboard/controllers/rgw_iam.py (hunk 1 above)
Comment (nmunet) [reply src/pybind/mgr/dashboard/controllers/rgw_iam.py:17]: ohh yess, I forgot to make this change
Thanks will update my PR.

Comment (nmunet) [reply src/pybind/mgr/dashboard/controllers/rgw_iam.py]: Fixed!

**Review by afreen23 — APPROVED** [review]

**Comment by afreen23** [comment]
@Pegonzal seeing some comments from your side. If this looks good to you we can merge, thanks.

**Review by Pegonzal — APPROVED** [review]
Thanks @nmunet PR LGTM

**Pull request merged**